packages/dataviews/src/dataviews-layouts/list/index.tsx

```diff
@@ -265,7 +254,9 @@ function ListItem< Item >( {
 						<div role="gridcell">
 							<DropdownMenu
 								trigger={
-									<CompositeItem
+									<Composite.Item
+										// @ts-expect-error Passing the store here is necessary
+										// in order to override the dropdown menu's store.
 										store={ store }
```

Comment:
For now, we have to continue passing explicitly a `store` object to this `Composite.Item` instance — without it, the the composite item is not included in the parent composite widget (see [conversation](https://github.com/WordPress/gutenberg/pull/63569#discussion_r1713359138)).

After some debugging, it looks like the explicit `store` override is necessary because otherwise this instance of `Composite.Item` gets the `DropdownMenu`'s store instead of the parent `Composite` store when reading the store from context.

@diegohaz , is this the expected behaviour (ie. passing `store` explicitly), or is this due to how `DropdownMenu` is implemented ([code](https://github.com/WordPress/gutenberg/blob/039bd9081abeb7ce93d7a57e56ca0abfbc6b7937/packages/components/src/dropdown-menu-v2/index.tsx#L309)) ?

Thank you!

Reply:
That's "expected" because the Ariakit Menu component also provides a Composite context. I acknowledge this is desirable in some cases but cumbersome in others, so this may change in the future.

Related issue: [https://github.com/ariakit/ariakit/issues/3754](https://github.com/ariakit/ariakit/issues/3754)

Reply:
Ah, that explains it! I didn't consider that different components could internally use another composite store instance.

I think that the best approach for us here is to use a custom `Composite` context to make sure that all sub-components consume the correct store (it's the same approach that we adopted in `Tabs` and `DropdownMenu`)

I tested those changes locally and they work well, and I opened a separate PR to land them: #64493

Once that PR is merged, I will update this PR (removing the explicit `store` prop), and we should be ready to land this one too.

Reply:
An internal context makes sense for passing down the store.  👍 

#64493 looks good! ✅ 

Noting that there appears to be one more instance where we explicitly pass `store` to a `Composite.Item`:

https://github.com/WordPress/gutenberg/blob/2c62c6aa2f30060b3acd40b225aebe4295ad0bf4/packages/block-directory/src/components/downloadable-block-list-item/index.js#L117

Might make sense to address it here before landing.

Reply:
I had somewhat missed that instance (and your comment). Thank you for flagging again